### ISSUE TITLE
[Docs/Website] Suggestions for small changes to the specification

### ISSUE THREAD

**Issue by deanwampler** [The-AI-Alliance/open-trusted-data-initiative#183]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Description

This is a follow up for #173, a list of "small" changes I recommend to the requirements, mostly clarifications and items that I recommend we no longer require for the reasons discussed (i.e., remove the checked box in the table for them).

For reference, here is the link to the [Hugging Face docs on dataset cards](https://huggingface.co/docs/datasets/dataset_card) that I'll refer to below as "HF docs".

See other sub-issues for additional recommendations.

### Affected Areas

- [x] Documentation files (docs/)
- [x] Website content (website/)

### Proposed Changes

1. `language-details` field: Two recommendations:
    a. The "HF docs" give examples from the https://en.wikipedia.org/wiki/IETF_language_tag, e.g., `en-US`. We should recommend people use only these tags. _However_, when I created the _static catalog_, I discussed that this field isn't very useful, because **every HF dataset we have cataloged has `en-US` as the sole value for this field!**
    b. Because it's not that useful, no longer make it required.
2. `pretty_name` field: Document that  is completely free form, but it can't be empty. See [https://github.com/Data-and-Trust-Alliance/DPS/blob/main/dps.txt](https://github.com/Data-and-Trust-Alliance/DPS/blob/648e07c8d86d8ac6aeeee703ab012bf77a1e0108/dps.txt#L512C1-L512C27), which inspired it.
3. `task_categories` field: 
    a. Provide easier to use documentation based on the [TypeScript code HF refers to](https://github.com/huggingface/huggingface.js/blob/main/packages/tasks/src/pipelines.ts) it could be better documented. 
    b. No longer make it required, though. While potentially very useful for searchers, when defined correctly, it shouldn't block consideration for our catalog.

**Comment by deanwampler** [comment]
More suggestions.

In the section [The Markdown Content in the Dataset Card](https://the-ai-alliance.github.io/open-trusted-data-initiative/dataset-requirements/#the-markdown-content-in-the-dataset-card):

1. `unique_metadata_identifier` field: Remove requirement to include it. While potentially very useful, it will exclude too many datasets. 
2. `dataset_sources` through `consent_documentation_location` fields: Consider how many of these can be enforced automatically in some way. If not, consider making them recommended vs. required.
3. `bias_risks_limitations` through `use_of_privacy_enhancing_technologies_pets` fields: Make optional, but recommended.
4. `dataset_card_authors` and `dataset_issue_date` fields: Nice to have, but they can probably be inferred from other metadata, so don't require them.
5. `range_dates_data_generation` field: Nice to have, but may be uncommon, so make it optional.

In the section [Derived Dataset Specification](https://the-ai-alliance.github.io/open-trusted-data-initiative/dataset-requirements/#derived-dataset-specification):

1. `unique_metadata_identifier` and `dataset_issue_date` fields: Make optional as in the previous section.

**Comment by jolson-ibm** [comment]
Same for `pretty_name`. There are 30,555 populated fields in a data set size of 435,719 (7%)

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/joeo/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/joeo/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


Count | pretty_name
-- | --
405141 | None
78 | BEIR Benchmark
48 | a
32 | test
31 | d
30 | s
30 | Evaluation run of google/gemma-2-2b
28 | general domain
23 |  
22 | test dataset
22 | Alpaca-Cleaned
21 | f
21 | Acronym Identification Dataset
17 | TOFU
17 | medical information retrieval
16 | e
16 | VCR
15 | information retrieval system for academic research papers
15 | sunny thakur
15 | tiny_demo
14 | BEIR-NL
14 | Buzz-V1.2
14 | t
13 | MSLR Shared Task
13 | GammaCorpus
12 | Su Id Asr
12 | OpenOrca
12 | Jv Id Asr
10 | Dolma
10 | SQuAD
10 | CMIRB Benchmark
10 | j
9 | MINT-1T
9 | Wikipedia
9 | MNIST
9 | pminervini_NQ_Swap
9 | Evaluation run of JalilH/gemma-2-2b
8 | R2MED Benchmark
8 | The Cross-lingual TRansfer Evaluation of Multilingual Encoders for Speech   (XTREME-S) benchmark is a benchmark designed to evaluate speech   representations across languages, tasks, domains and data regimes. It covers   102 languages from 10+ language families, 3 different domains and 4 task   families: speech recognition, translation, classification and retrieval.
8 | IMDB
8 | p
8 | OpenVid-1M
8 | medical information search
8 | MMLU loader with no auxiliary train set
8 | sentiment-classification-reviews-with-drift
8 | c
8 | academic research data retrieval
8 | SQuAD for question generation
8 | 'Invalid license_name'
7 | RNAcentral
7 | fusechat_mixture
7 | quran_qari_akhdar
7 | SemEval2012 task 2 Relational Similarity
7 | T
7 | r
7 | g
7 | Multi-XScience
7 | v
7 |  
7 | Common Task-Oriented Dialogues in Taiwan
7 | Multi-News
7 | LibriSpeech
7 | i
7 | Numina-Olympiads
6 | Pokémon BLIP captions
6 | HSSD
6 | The Expresso Dataset
6 | service search for translation and editing
6 | CNN / Daily Mail
6 | Test
6 | D
6 | No Robots
6 | ImageNet
6 | WCEP-10
6 | VideoGameBunny Dataset
6 | quickb-qa
6 | Grade School Math 8K
6 | OpenHermes 2.5
6 | S
6 | AI vs Human CNN Daily News
6 | DIALOGSum Corpus
6 | h
6 | Measuring Massive Multitask Language Understanding
6 | MMEB
6 | health insurance information
6 | Genshin Voice
6 | dataset
6 | TyDi QA
6 | medical domain
6 | TruthfulQA
6 | Multilingual Grade School Math Benchmark (MGSM)
5 | Instruction Tuning with GPT-4
5 | R
5 | sql-create-context
5 | GLUE (General Language Understanding Evaluation benchmark)
5 | Evaluation run of richmondsin/finetuned-gemma-2-2b-output-layer-20-4k-0
5 | Tatoeba
5 | PubMed
5 | Evaluation run of meta-llama/Llama-3.1-8B-Instruct
5 | Natural Reasoning



</body>

</html>

**Comment by jolson-ibm** [comment]
Finally: task_categories. 48,973 data sets with values (11.2%) This is another long tail one . First 100 by count:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/joeo/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/joeo/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


Count | task_categories
-- | --
386737 | None
10848 | ['robotics']
5207 | ['text-generation']
4319 | ['text-classification']
3859 | ['text-to-image']
2913 | ['question-answering']
1969 | ['translation']
1599 | ['text-retrieval']
924 | ['image-classification']
776 | ['token-classification']
622 | ['image-to-text', 'text-to-image']
610 | ['automatic-speech-recognition']
595 | ['text2text-generation']
560 | ['text-classification', 'token-classification', 'question-answering',   'summarization', 'text-generation']
513 | ['summarization']
454 | ['object-detection']
406 | ['feature-extraction', 'sentence-similarity']
384 | ['question-answering', 'text-generation']
366 | ['image-to-text']
362 | ['visual-question-answering']
304 | ['text-generation', 'question-answering']
291 | ['text-generation', 'text2text-generation']
290 | ['image-segmentation']
280 | ['feature-extraction']
271 | ['text-generation', 'fill-mask']
256 | ['text-generation', 'text2text-generation', 'question-answering']
221 | ['conversational']
204 | ['other']
193 | ['text-to-speech']
192 | ['sentence-similarity']
187 | ['audio-classification']
185 | ['table-question-answering']
178 | ['text-classification', 'text-generation']
172 | ['tabular-classification']
153 | ['image-to-image']
129 | ['text-classification', 'image-classification', 'feature-extraction',   'image-segmentation', 'image-to-image', 'image-to-text', 'object-detection',   'summarization', 'zero-shot-image-classification']
113 | ['time-series-forecasting']
110 | ['text-generation', 'text-classification']
106 | ['graph-ml']
105 | ['multiple-choice']
93 | ['automatic-speech-recognition', 'text-to-speech']
92 | ['video-classification']
89 | ['reinforcement-learning']
86 | []
80 | ['image-text-to-text']
79 | ['text-classification', 'token-classification']
77 | ['text-generation', 'table-question-answering', 'summarization',   'text-retrieval', 'question-answering', 'text-classification']
76 | ['text-classification', 'question-answering']
76 | ['zero-shot-classification']
74 | ['text2text-generation', 'text-generation']
72 | ['tabular-regression']
62 | ['multiple-choice', 'question-answering', 'zero-shot-classification',   'text2text-generation', 'table-question-answering', 'text-generation',   'text-classification', 'tabular-classification']
61 | ['text-to-image', 'image-to-text']
59 | ['text-classification', 'zero-shot-classification']
59 | ['text-classification', 'text-retrieval']
59 | ['visual-question-answering', 'question-answering']
58 | ['summarization', 'text-generation']
58 | ['question-answering', 'text2text-generation']
58 | ['automatic-speech-recognition', 'audio-classification']
57 | ['document-question-answering', 'visual-document-retrieval']
55 | ['text-classification', 'feature-extraction']
55 | ['video-text-to-text']
53 | ['conversational', 'text-generation']
53 | ['question-answering', 'multiple-choice']
52 | ['question-answering', 'summarization', 'text-generation']
52 | ['question-answering', 'summarization']
51 | ['audio-to-audio']
50 | ['question-answering', 'text-generation', 'text2text-generation']
50 | ['machine-translation']
50 | ['summarization', 'text2text-generation']
49 | ['text-retrieval', 'zero-shot-retrieval', 'information-retrieval',   'zero-shot-information-retrieval']
49 | ['speech-recognition']
48 | ['text-generation', 'text2text-generation', 'text-retrieval',   'question-answering', 'sentence-similarity']
46 | ['text-generation', 'conversational']
44 | ['conditional-text-generation']
42 | ['text-to-video']
42 | ['fill-mask']
41 | ['translation', 'text-generation']
41 | ['text-generation', 'text2text-generation', 'text-retrieval',   'question-answering']
41 | ['question-answering', 'text-classification']
37 | ['text-to-image', 'image-to-image']
37 | ['multiple-choice', 'question-answering']
36 | ['self-supervised-pretraining']
35 | ['text-to-speech', 'automatic-speech-recognition']
35 | ['text-ranking']
34 | ['text-classification', 'summarization']
34 | ['translation', 'text2text-generation']
34 | ['text-classification', 'text2text-generation']
34 | ['image-classification', 'image-segmentation']
33 | ['sentiment-analysis']
33 | ['image-feature-extraction']
31 | ['question-answering', 'table-question-answering']
31 | ['image-to-3d']
30 | ['text-classification', 'question-answering', 'text-generation']
28 | ['fill-mask', 'text-generation']
28 | ['question-answering', 'conversational']
27 | ['text-generation', 'summarization']
27 | ['question-answering', 'visual-question-answering']
26 | ['table-question-answering', 'question-answering']
26 | ['visual-question-answering', 'image-to-text']



</body>

</html>
